### ISSUE TITLE
bug: hostmetrics receiver generating excessive log spam

### ISSUE THREAD

**Issue by pratheekrebala** [GoogleCloudPlatform/ops-agent#249]
Currently, the hostmetrics receiver generates a lot of log spam because of a permissions issue preventing the collector from accessing the path of a process owned by another user. An upstream bug has been reported at:

- https://github.com/open-telemetry/opentelemetry-collector/issues/3004

This has been [marked as non-harmful log spam](https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting#log-spam) which is technically true, but is making logs very hard to use since these messages constitute a majority of my logs. I was wondering if it would be possible to set a filter for these types of logs at source? I am currently filtering out all logs from the opentelemetry agent to make my logs usable. This isn't ideal but feels like the best option until it can be addressed upstream.

I think it would be great this could be added to the Fluentbit config or if it would be possible to specify exclude filters for logs through configuration.

```
[FILTER]
    name    grep
    match    *
    exclude msg otelopscol
``` 

Since there isn't an easy way to specify the above filter as a configuration option, I'm manually adding the following filter to my rsyslog configuration. 

`/etc/rsyslog.d/99-exclude-otel.conf`:
```
if ($programname contains "otelopscol") then {                   
   action(type="omfile" file="/var/log/dropped-msgs.log")
   stop
}
```

**Comment by shuLhan** [comment]
For anyone who build ops-agent manually (or running Arch), here is [1] the patch to minimize this logging spam.

```
diff --git a/apps/hostmetrics.go b/apps/hostmetrics.go
index 84bdc49..14a77f7 100644
--- a/apps/hostmetrics.go
+++ b/apps/hostmetrics.go
@@ -43,8 +43,6 @@ func (r MetricsReceiverHostmetrics) Pipelines() []otel.Pipeline {
 					"filesystem": struct{}{},
 					"network":    struct{}{},
 					"paging":     struct{}{},
-					"process":    struct{}{},
-					"processes":  struct{}{},
 				},
 			},
 		},
diff --git a/confgenerator/confgenerator.go b/confgenerator/confgenerator.go
index 1916bcd..a8443ec 100644
--- a/confgenerator/confgenerator.go
+++ b/confgenerator/confgenerator.go
@@ -47,7 +47,7 @@ func (uc *UnifiedConfig) GenerateOtelConfig(hostInfo *host.InfoStat) (string, er
 	}.Pipeline()
 
 	if uc.Metrics.Service.LogLevel == "" {
-		uc.Metrics.Service.LogLevel = "info"
+		uc.Metrics.Service.LogLevel = "error"
 	}
 	otelConfig, err := otel.ModularConfig{
 		LogLevel:  uc.Metrics.Service.LogLevel,
@@ -126,7 +126,7 @@ func (uc *UnifiedConfig) GenerateFluentBitConfigs(logsDir string, stateDir strin
 func (l *Logging) generateFluentbitComponents(userAgent string, hostInfo *host.InfoStat) ([]fluentbit.Component, error) {
 	var out []fluentbit.Component
 	if l.Service.LogLevel == "" {
-		l.Service.LogLevel = "info"
+		l.Service.LogLevel = "error"
 	}
 	out = append(out, fluentbit.Service{LogLevel: l.Service.LogLevel}.Component())
 ```

Its very annoying.

[1] https://aur.archlinux.org/cgit/aur.git/tree/0001-apps-hostmetrics.patch?h=google-cloud-ops-agent-git

**Comment by KDanisme** [comment]
Seems like they patched it in this [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7176)
Is it possible to bump the opentelemetry-collector-contrib version?

**Comment by braydonk** [comment]
The problem outlined in this issue should be fixed in Ops Agent 2.10 and above. There can be scenarios where similar log spam from `hostmetrics` can occur, but it won't be occurring by default with processes that it shouldn't be trying to read in the first place.

If you encounter something similar to this issue in a new version of the Ops Agent, please open a new issue that we can triage separately.
